packages/Popover/src/Popover.js

```diff
@@ -0,0 +1,517 @@
+    const { withoutPortal } = this.props;
+    const top = withoutPortal
+      ? this.calcWithoutPortalTop({ justification })
+      : this.calcTop({ justification });
+    const left = withoutPortal
+      ? this.calcWithoutPortalLeft({ justification })
+      : this.calcLeft({ justification });
+
+    if (['top', 'bottom', 'center-vertical'].includes(justification)) {
+      return this.checkVerticalWindowCollision(top);
+    }
+
+    if (['left', 'right', 'center-horizontal'].includes(justification)) {
+      return this.checkHorizontalWindowCollision(left);
+    }
+  }
+
+  // Check if horizontal position collides with edge of window
+  checkHorizontalWindowCollision(left) {
+    const contentRefPos = this.getRefPosition(this.contentRef);
+    const tooWide = left + contentRefPos.width > this.state.windowWidth;
+
+    return !(left < 0 || tooWide);
+  }
+
+  // Check if vertical position collides with edge of window
+  checkVerticalWindowCollision(top) {
+    const contentRefPos = this.getRefPosition(this.contentRef);
+    const tooTall = top + contentRefPos.height > this.state.windowHeight;
+
+    return !(top < 0 || tooTall);
+  }
+
+  // Returns the 'top' position in pixels for a valid alignment or justification.
+  calcTop({ alignment, justification }) {
+    const { referenceElPos } = this.state;
+    const contentRefPos = this.getRefPosition(this.contentRef);
+
+    if (alignment) {
+      switch (alignment) {
+        case 'top':
+          return referenceElPos.top - contentRefPos.height;
+
+        case 'bottom':
+          return referenceElPos.top + referenceElPos.height;
+      }
+    }
+
+    if (justification) {
+      switch (justification) {
+        case 'top':
+          return referenceElPos.top;
+
+        case 'bottom':
+          return (
+            referenceElPos.top + referenceElPos.height - contentRefPos.height
+          );
+
+        case 'center-vertical':
+          return (
+            referenceElPos.top +
+            referenceElPos.height / 2 -
+            contentRefPos.height / 2
+          );
+      }
+    }
+  }
+
+  calcWithoutPortalTop({ alignment, justification }) {
+    const contentRefPos = this.getRefPosition(this.contentRef);
+
+    if (alignment) {
+      switch (alignment) {
+        case 'top':
+          return `${0 - contentRefPos.height}px`;
+
+        case 'bottom':
+          return '100%';
+      }
+    }
+
+    if (justification) {
+      switch (justification) {
+        case 'top':
+          return 0;
+
+        case 'bottom':
+          return `calc(100% - ${contentRefPos.height}px)`;
+
+        case 'center-vertical':
+          return `calc(50% - ${contentRefPos.height / 2}px)`;
+      }
+    }
+  }
+
+  // Returns the 'left' position in pixels for a valid alignment or justification.
+  calcLeft({ alignment, justification }) {
+    const { referenceElPos } = this.state;
+    const contentRefPos = this.getRefPosition(this.contentRef);
+
+    if (alignment) {
```

Comment:
We don't need these conditionals before the switches. It will work just fine without. Same for the justification switch.